File: abc312a/abc312a.hs
import Control.Applicative
import Data.List

main :: IO ()
main = solve <$> getLine >>= putStrLn

solve :: String -> String
solve = maybe "No" (const "Yes") . flip find ls . (==)
  where
    ls = ["ACE", "BDF", "CEG", "DFA", "EGB", "FAC", "GBD"]

File: abc312a/abc312a.ml
open Printf
open Scanf

let solve s =
  let l = ["ACE"; "BDF"; "CEG"; "DFA"; "EGB"; "FAC"; "GBD"] in
  match List.find_opt (fun x -> x = s) l with
    Some _ -> "Yes"
  | None -> "No"

let () =
  scanf "%s " solve |> printf "%s\n"
